core-bundle/src/Resources/contao/templates/backend/be_login.html5

```diff
@@ -36,7 +36,7 @@
   <?php $this->endblock(); ?>
 
 </head>
-<body class="<?= $this->ua ?>">
+<body class="<?= $this->ua ?>"<?= $this->dataAttributes ?>>
```

Comment:
Could this be an array instead and mapped+imploded in place?

Reply:
Yeah, was not sure about more logic in the templates or in the method `addBackendConfig`

Reply:
It's not really nice but imo this is exactly what templates should be doing. In Twig you would probably write a macro or custom function (see image studio macros for instance), in Contao templates you should probably doing something similar or just stick to one of these:

```suggestion
<body class="<?= $this->ua ?>"<?= implode(' ', array_map(static function($k, $v){ return "$k=\"$v\""; }, array_keys($this->dataAttributes), $this->dataAttributes)) ?>>
```

```suggestion
<body class="<?= $this->ua ?>"<? foreach($this->dataAttributes as $name => $value): ?> <?= $name ?>="<?= $value?>"<?php endforeach; ?>>
```

But that's only my opinion of course. 🙂

Reply:
An array has the big benefit that you're not tied to the exact output or even data-attributes.

Reply:
Yes, I would also prefer the logic in the template. Then you can filter or do something else in the template, if you want.
But would you then define the attributes with prefix `data-` in the configuration e.g. `data-app-name: 'My App'` or in the method `addBackendConfig` or in the template???

Reply:
> But would you then define the attributes with prefix data- in the configuration

Yes, because then you could even go for an existing attribute. Like `class` or something :smiley: .

Reply:
But `class` is allready defined for ua...

Reply:
You could add it to the other body attributes, though. Not sure about this.

Reply:
Okay for now I refactored the attributes without prefix + loop logic in the templates!